spring-statemachine-core/src/main/java/org/springframework/statemachine/config/configuration/StateMachineFactoryConfiguration.java

```diff
@@ -188,13 +179,6 @@ public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
 			this.beanFactory = beanFactory;
 		}
 
-		@Autowired(required=false)
```

Comment:
It's actually interesting idea to try to use `beanFactory` directly instead of letting context to autowire these configurers. As this was a concept copied from spring security and it relied on a fact that `autowiring` happens before `afterPropertiesSet` is getting called.